core/src/components/segment-button/segment-button.ionic.scss

```diff
@@ -67,7 +67,8 @@
 // --------------------------------------------------
 
 :host(.segment-button-checked) ::slotted(ion-label),
-:host(.segment-button-checked) ::slotted(ion-icon) {
+:host(.segment-button-checked) ::slotted(ion-icon),
+:host(.segment-button-checked) ::slotted(*) {
```

Comment:
You should remove the redundant selectors, since the star selector will select ion-label and ion-icon already. You should probably document what you're trying to select here, though.